python/dlisio/plumbing/errorhandler.py

```diff
@@ -0,0 +1,216 @@
+import logging
+from enum import Enum
+
+from .. import core
+
+class ErrorHandler(core.error_handler):
+    """ Defines rules about error handling
+
+    Many .dlis files happen to be not compliant with specification or simply
+    broken. This class gives user some control over handling of such files.
+
+    When dlisio encounters a specification violation, it categories the issue
+    based on the severity of the violation. Some issues are easy to ignore
+    while other might force dlisio to give up on its current task.
+    ErrorHandler supplies an interface for changing how dlisio reacts to
+    different violation in the file.
+
+    Different categories are *info*, *minor*, *major* and *critical*:
+
+    ======== ===================================================================
+    Severity Description
+    ======== ===================================================================
+    critical Any issue that forces dlisio stop its current objective prematurely
+             is categorised as critical.
+
+             By default a critical error raises a RuntimeError.
+
+             An example would be file indexing, which happens at load. Suppose
+             the indexing fails midways through the file. There is no way for
+             dlisio to reliably keep indexing the file.
+             However, it is likely that the file is readable up until the point
+             of failure. Changing the behaviour of critical from raising an
+             Exception to logging would in this case mean that a partially
+             indexed file is returned by load.
+    major    Any issue that dlisio is able to recover from under heavy
```

Comment:
> heavy developer's assumption

Generally we refer to dlisio as the entity in docs, not to the developers. It's implied that any behaviour dlisio has is decided by developers. I also generally try to avoid formulating sentences such that I have to use the word `user`

Also, heavy assumptions. That sounds frightening, and the definition of `heavy` in this case is up for grabs. rather than trying to define heavy and light I would try to explain the behaviour of dlisio when a major/minor issue is encountered.

major:

These are a result of direct spec violations in the file, but where dlisio makes an assumption on what the broken information [1] _should_ have been, and continues parsing the file on this assumption. If no other major or critical issues are reported, it's likely that the assumption where correct and that dlisio parsed the file correctly. However, no guarantees can be made.

[1] Note that "information" in this case refers to data in the file that tells dlisio how the file should be parsed. Not to the actual parsed data.

Minor

Like, Major these are also a result of direct spec-violations, and dlisio makes similar assumptions to keep parsing the file. Minor issues are generally less severe and easier to handle correctly. However, no guarantees can be made about the parsed data